app/models/pawn.rb

```diff
@@ -1,2 +1,22 @@
 class Pawn < Piece
-end
\ No newline at end of file
+  def valid_move?(new_rank, new_file)
+    rank_diff = new_rank - rank
+    file_diff = new_file - file
+
+    return false if is_move_malformed?(new_rank, new_file) || is_move_obstructed?(new_rank, new_file)
+
+    if color == "white"
+      # Standard pawn move
+      return true if rank_diff == 1 && file_diff == 0
+      # Double move
+      return true if rank == 2 && rank_diff == 2 && file_diff == 0
+      # Capture move
+      return true if rank_diff == 1 && file_diff.abs == 1 && capture_move?(new_rank, new_file)
+    else
+      return true if rank_diff == -1 && file_diff == 0
+      return true if rank == 7 && rank_diff == -2 && file_diff == 0
+      return true if rank_diff == -1 && file_diff.abs == 1 && capture_move?(new_rank, new_file)
```

Comment:
Little math trick that may help dedup a bit here, you can set a modifier variable then multiply against that to create your positive negatives. X * 1 is always X, and X * -1 is always -X. 

With that and a var for your rank position, all of the returns could then be unified wiht those.